t/01-temp-files.t

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/perl
+# This program is open source, licensed under the PostgreSQL License.
+# For license terms, see the LICENSE file.
+#
+# Copyright (C) 2012-2022: Open PostgreSQL Monitoring Development Group
+
+use strict;
+use warnings;
+
+use lib 't/lib';
+use pgNode;
+use pgSession;
+use File::Path qw(rmtree);
+use Test::More tests => 39;
+
+my $node = pgNode->get_new_node('prod');
+my $proc;
+
+$node->init;
+$node->start;
+
+### Beginning of tests ###
+
+# This service can run without thresholds
+
+# Tests for PostreSQL 8.1 and before
+SKIP: {
+    skip "testing incompatibility with PostgreSQL 8.1 and before", 3
+        if $node->version >= 8.2;
+
+    $node->command_checks_all( [
+        './check_pgactivity', '--service'  => 'temp_files',
+                              '--username' => getlogin,
+                              '--format'   => 'human',
+        ],
+        1,
+        [ qr/^$/ ],
+        [ qr/^Service temp_files is not compatible with host/ ],
+        'non compatible PostgreSQL version'
+    );
+}
+
+SKIP: {
+    skip "incompatible tests with PostgreSQL < 8.2", 34 if $node->version < 8.2;
+
+    # basic check
+    $node->command_checks_all( [
+        './check_pgactivity', '--service'  => 'temp_files',
+                              '--username' => getlogin,
+                              '--format'   => 'human',
+                              '--dbname'   => 'template1',
+        ],
+        0,
+        [ qr/^Service  *: POSTGRES_TEMP_FILES$/m,
+          qr/^Returns  *: 0 \(OK\)$/m,
+          qr/^Message  *: 3 tablespace\(s\)\/database\(s\) checked$/m,
+        ],
+        [ qr/^$/ ],
+        'basic check'
+    );
+    sleep 2;
```

Comment:
The `sleep` calls surely need some comments to explain why it might fails without it. It will give a hint to future devs as well :)

Reply:
```suggestion
   # The added sleep ensures that two tests are not executed within the same seconds. 
   # The time difference is used to compute the Fpm and Bpm perfstat, if it's zero the 
   # check crashes (division by zero).
    sleep 2;
```
@ioguix : is this one fine ?